django_keyerror/error.py

```diff
@@ -20,6 +20,10 @@ def __init__(self, exc_type, exc_value, exc_traceback, ident=None):
         tb = traceback.extract_tb(exc_traceback)
         synopsis = traceback.format_exception_only(exc_type, exc_value)[-1]
 
+        cause = getattr(exc_value, '__cause__', None)
+        if cause is not None:
+            tb += traceback.extract_tb(cause.__traceback__)
```

Comment:
Do we want to have a "caused by.." extra text here? Just wondering what it looks like…

Reply:
I'm not sure. I was aiming for a solution which didn't involve any changes to the server side.

Here's an example of the traceback as rendered on KeyError:
```
Traceback (most recent call last):
  File "test_keyerror.py", line 19, in main
    ok_function()
  File "test_keyerror.py", line 14, in ok_function
    bad_function()
  File "/usr/lib/python3.5/contextlib.py", line 77, in __exit__
    self.gen.throw(type, value, traceback)
  File "/home/peter/errors.py", line 87, in error_wrapper
    raise MyException("Testing error 3") from e
  File "/home/peter/errors.py", line 85, in error_wrapper
    yield
  File "test_keyerror.py", line 14, in ok_function
    bad_function()
  File "test_keyerror.py", line 9, in bad_function
    ... . wat
errors.MyException: Testing error 3
```

It's a little bit odd that the traceback carries straight on through the re-raise (`raise MyException(comment) from e`) and then repeats part of the stack (`File "test_keyerror.py", line 14, in ok_function`) though not too bad as a first pass.

Reply:
Hm, thanks. Let's go with it,.